### PR TITLE
Add LossLink node.

### PR DESCRIPTION
**Pull request by jetuk**

### CODE REVIEW

File: pywr/nodes.py [hunk 1]
```diff
@@ -1074,4 +1074,96 @@ def after(self, timestep):
...
+    A fixed proportional loss of all flow through this node is sent to an internal output node. Max and min flows
+    applied to this node are enforced on the net output after losses. The node itself records the net output
+    in its flow attribute (which would be used by any attached recorders).
+
+    Parameters
+    ----------
+    model : `pywr.model.Model`
+    name : string
+        Name of the node.
+    loss_factor : float
```
Comment (jetuk) [inline pywr/nodes.py:1089]: This probably should have a way of specifying the loss_factor based on gross inflow or net outflow, and constraints on gross inflow and/or net outflow.

File: tests/test_run.py [hunk 2]
```diff
@@ -672,6 +672,24 @@ def test_breaklink_node():
     assert_allclose(transfer.storage.volume, 0)
 
 
+def test_loss_link_node():
```
[the remark this answers is not in the document]
Comment (Batch21) [reply tests/test_run.py:675]: and 1.0

File: pywr/nodes.py (hunk 1 above)
Comment (Batch21) [reply pywr/nodes.py:1089]: any plans to still do this?

Comment (jetuk) [reply pywr/nodes.py:1089]: I'm not sure it's too tricky? What do you think?

Comment (Batch21) [reply pywr/nodes.py:1089]: probably worth doing if its not too tricky. Does it just involve changing which component node commits flow to the parent node?

Comment (jetuk) [reply pywr/nodes.py:1089]: I think you have to change which nodes are in the internal `AggregatedNode`. Or perhaps just the calculation of the factor.

Comment (jetuk) [reply pywr/nodes.py:1089]: Let's leave it be for now.

**Review by Batch21 — APPROVED** [review]

**Pull request merged**